docs/errors.md

```diff
@@ -0,0 +1,231 @@
+
+- #### `MongoResourceClosedError`
+
+  - Thrown when there is an attempt to access a resource which has already been or will be closed/destroyed.
+    - Children of this error class include:
+      - **`MongoServerClosedError`**: Thrown when an attempt is made to operate on a closed server.
+      - **`MongoStreamClosedError`**: Thrown when an attempt is made to operate on a closed stream.
+      - **`MongoTopologyClosedError`**: Thrown when an attempt is made to operate on a dropped, or otherwise unavailable, database.
+
+- #### `MongoCursorError`
+
+  - Thrown when the user incorrectly uses a cursor object.
+    - Children of this error class include:
+      - **`MongoInvalidCursorOperationError`**: Thrown when the user calls a function or method that is not supported on the current cursor.
+      - **`MongoCursorExhaustedError`**: Thrown when an attempt is made to read from a cursor that has been exhausted.
+
+- #### `MongoStreamError`
+
+  - Thrown when a stream operation fails to execute.
+  - Children of this error class include:
+    - **`MongoChangeStreamError`**: Thrown when an error is encountered when operating on a ChangeStream.
+    - **`MongoGridFSStreamError`**: Thrown when an unexpected state is reached when operating on a GridFSStream.
+    - **`MongoGridFSChunkError`**: Thrown when a malformed or invalid chunk is encountered when reading from a GridFSStream.
+
+- #### `MongoBatchReExecutionError`
+
+  - Thrown when a user attempts to reexecute a batch command when one of the constituent commands has failed.
+
+- #### `MongoServerSelectionError`
+
+  - Thrown when the driver fails to select a server to complete an operation.
+
+- #### `MongoInvalidClassInstantiationError`
+
+  - Thrown when an internal class that should not be instantiated is directly instantiated.
+
+## `MongoNetworkError`
+
+These are errors encountered at runtime which occur when the driver encounters an issue in the network which leads to an inability to connect to a mongo server instance. Children of this class include:
+
+- #### `MongoNetworkTimeoutError`
+
+  - Thrown when a timeout expires in attempting to connect to the mongo server
+
+## `MongoServerError`
+
+These are errors which wrap error responses received from the server.
+
+## `MongoSystemError`
+
+These are errors which originate from faulty environment setup.
+
+# Test Plan
+
+The test plan consists of a series of prose tests.
+As numerous errors are being introduced, select classes will be tested.
+The classes to be tested will be selected based on two characteristics:
+
+1. The **frequency** with which users may encounter this error. Errors that users will likely catch in their code, including but not limited to `MongoIOError` and `MongoNetworkTimeoutError`, are part of the test plan. _Note:_ Error classes that should never be instantiated, such as `MongoLogicError` and `MongoRuntimeError`, will not be tested as the user should not encounter them.
```

Comment:
MongoIOError has been removed, you can hold off making the change right now if you want lets use this comment as an reminder to comb over the docs for dropped errors and to update the mermaid diagrams

Reply:
Everything has been updated 😎